### PR TITLE
Fix create pr add build image tags

### PR DESCRIPTION
**Pull request by kentsanggds**
- Fix the create PR so that the branch name is used for checking out github and creating the PR
- also rename ghcr to build-image to make it clearer 
- and add in a build image for the tag version so that images can be quickly identified to a tagged release

### CODE REVIEW

File: .github/workflows/build-image-on-tags.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+
+jobs:
+  build_and_push:
+    name: ${{ matrix.app.name }} (${{ matrix.arch }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (sengi) [inline .github/workflows/build-image-on-tags.yaml:13]: Unusual use of matrix builds, but no probs :)

File: docker/create-pr.sh [hunk 2]
```diff
@@ -29,6 +30,6 @@ for ENV in $(echo $ENVS | tr "," " "); do
     done
     git commit -m "Update image tags for ${ENV} to ${IMAGE_TAG}"
     git push --set-upstream origin "${BRANCH}"
-    gh pr create --title "Update image tags for ${ENV} (${IMAGE_TAG})" --base main --head "${BRANCH}-${ENV}" --fill
+    gh pr create --title "Update image tags for ${ENV} (${IMAGE_TAG})" --base main --head "${BRANCH}" --fill
```
Comment (sengi) [inline docker/create-pr.sh:33]: You might find the PR workflow for automated commits a little tricky. For the rest of GOV.UK we just commit and merge directly. No problem if you wanna experiment with this though.

**Review by sengi — APPROVED** [review]
I suspect there are actions and/or workflows you could reuse from [govuk-infrastructure](https://github.com/alphagov/govuk-infrastructure/tree/main/.github).

I should have thought to suggest this earlier (my bad) — but worth chatting with @theseanything about the whole CI/CD / build/release automation thing as he might have suggestions that could save you a lot of hassle.

File: docker/create-pr.sh (hunk 2 above)
Comment (kentsanggds) [reply docker/create-pr.sh:33]: I think lets try it out first, it will probably be mainly me who will be running the process

**Pull request merged**